### PR TITLE
Setup Git server and integration test

### PR DESCRIPTION
**Pull request by junha1**

### CODE REVIEW

File: repository/tests/integration_test.rs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+use simperby_network::{NetworkConfig, Peer};
+use simperby_repository::{raw::*, *};
+use simperby_test_suite::*;
+
+#[ignore]
+#[tokio::test]
+async fn basic_1() {
+    let port = 8801;
+    let n = 4;
+    let (rs, keys) = generate_standard_genesis(n);
```
Comment (TomTaehoonKim) [inline repository/tests/integration_test.rs:11]:
```suggestion
    let (rs, keys) = generate_standard_genesis(4);
```

File: repository/tests/integration_test.rs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    setup_pre_genesis_repository(&client_node_dir, rs.clone()).await;
+    let mut client_node_repo = DistributedRepository::new(
+        RawRepositoryImpl::open(&client_node_dir).await.unwrap(),
+        config,
+    )
+    .await
+    .unwrap();
+    client_node_repo.genesis().await.unwrap();
+
+    // Step 0: create an agneda and let the client update that
```
Comment (TomTaehoonKim) [inline repository/tests/integration_test.rs:54]:
```suggestion
    // Step 0: create an agenda and let the client update that
```

File: repository/tests/integration_test.rs [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        .unwrap();
+    assert_eq!(
+        client_node_repo.get_agendas().await.unwrap(),
+        vec![(agenda_commit, agenda.to_hash256())]
+    );
+    server_node_repo
+        .approve(
+            &agenda_commit,
+            keys.iter()
+                .map(|(_, k)| TypedSignature::sign(&agenda, k).unwrap())
```
Comment (TomTaehoonKim) [inline repository/tests/integration_test.rs:71]:
```suggestion
                .map(|(_, private_key)| TypedSignature::sign(&agenda, private_key).unwrap())
```

File: repository/tests/integration_test.rs [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        .unwrap();
+    assert_eq!(
+        client_node_repo.get_blocks().await.unwrap(),
+        vec![(block_commit, block.to_hash256())]
+    );
+
+    // Step 2: finalize a block and let the client update that
+    let block_proof = keys
+        .iter()
+        .map(|(_, k)| TypedSignature::sign(&block, k).unwrap())
```
Comment (TomTaehoonKim) [inline repository/tests/integration_test.rs:94]: ditto

File: test-suite/Cargo.toml [hunk 5]
```diff
@@ -15,3 +15,5 @@ futures = "0.3"
 log = "0.4"
 simperby-node = { path = "../node" }
 rand = "0.8"
+path-slash = "0.2.1"
+tempfile = "3"
```
Comment (TomTaehoonKim) [inline test-suite/Cargo.toml:19]:
```suggestion
tempfile = "3"

```

File: test-suite/src/lib.rs [hunk 6]
```diff
@@ -1,7 +1,10 @@
+use path_slash::PathExt as _;
 use simperby_node::simperby_common::*;
+use tempfile::TempDir;
 
 #[cfg(target_os = "windows")]
 pub async fn run_command(command: impl AsRef<str>) {
+    println!("> RUN: {}", command.as_ref());
```
Comment (TomTaehoonKim) [inline test-suite/src/lib.rs:7]:
```suggestion
```

File: test-suite/src/lib.rs [hunk 7]
```diff
@@ -14,6 +17,7 @@ pub async fn run_command(command: impl AsRef<str>) {
 
 #[cfg(not(target_os = "windows"))]
 pub async fn run_command(command: impl AsRef<str>) {
+    println!("> RUN: {}", command.as_ref());
```
Comment (TomTaehoonKim) [inline test-suite/src/lib.rs:20]:
```suggestion
```

File: repository/tests/integration_test.rs [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+    setup_pre_genesis_repository(&server_node_dir, rs.clone()).await;
+    let mut server_node_repo = DistributedRepository::new(
+        RawRepositoryImpl::open(&server_node_dir).await.unwrap(),
+        config.clone(),
+    )
+    .await
+    .unwrap();
+    server_node_repo.genesis().await.unwrap();
+
+    tokio::spawn(async move { server::run_server(&server_node_dir, 7301).await });
```
Comment (TomTaehoonKim) [inline repository/tests/integration_test.rs:41]: 7301 is default port for what?

File: test-suite/src/lib.rs (hunk 7 above)
Comment (junha1) [reply test-suite/src/lib.rs:20]: why?

File: repository/tests/integration_test.rs (hunk 8 above)
Comment (junha1) [reply repository/tests/integration_test.rs:41]: It's not something default. It's just an arbitrary number for the test.

File: test-suite/src/lib.rs (hunk 7 above)
Comment (TomTaehoonKim) [reply test-suite/src/lib.rs:20]: I thought it was just for you running the test locally

Comment (TomTaehoonKim) [reply test-suite/src/lib.rs:20]: junha's reply: it's a feature to show which command it was for the following results of the command

**Review by TomTaehoonKim — APPROVED** [review]

File: test-suite/src/lib.rs [hunk 9]
```diff
@@ -78,14 +82,32 @@ pub fn generate_standard_genesis(
     )
 }
 
+/// Creates a `repository` directory inside the given directory
+/// and initilaize a pre-genesis repository.
```
Comment (JeongHunP) [inline test-suite/src/lib.rs:86]:
```suggestion
/// and initialize a pre-genesis repository.
```

**Pull request merged**